cookbooks/bcpc-hadoop/attributes/yarn.rb

```diff
@@ -18,8 +18,16 @@
 yarn_pid_dir = '/var/run/hadoop-yarn'
 yarn_user = 'yarn'
 yarn_conf_dir = '/etc/hadoop/conf'
-yarn_logfile = 'yarn-$(hostname).log'
 yarn_policyfile = 'hadoop-policy.xml'
+yarn_logfile = 
+  if node.run_list.expand(node.chef_environment)
+         .recipes.include?('bcpc-hadoop::datanode')
+      'yarn-yarn-nodemanager-$(hostname).log'
```

Comment:
I am not sure if we need this. We don't specify log file names for any other service.